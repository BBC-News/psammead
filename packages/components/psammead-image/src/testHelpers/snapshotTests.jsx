import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { custom, landscape, portrait, square } from './fixtureData';

const snapshotTests = (Component, additionalProps) => {
  shouldMatchSnapshot(
    'should render landscape image correctly',
    <Component
      alt={landscape.alt}
      attribution={landscape.attribution}
      src={landscape.src}
      height={landscape.height}
      width={landscape.width}
      {...additionalProps}
    />,
  );
  shouldMatchSnapshot(
    'should render portrait image correctly',
    <Component
      alt={portrait.alt}
      attribution={portrait.attribution}
      src={portrait.src}
      height={portrait.height}
      width={portrait.width}
      {...additionalProps}
    />,
  );
  shouldMatchSnapshot(
    'should render square image correctly',
    <Component
      alt={square.alt}
      attribution={square.attribution}
      src={square.src}
      height={square.height}
      width={square.width}
      {...additionalProps}
    />,
  );
  shouldMatchSnapshot(
    'should render image with custom dimensions correctly',
    <Component
      alt={custom.alt}
      attribution={custom.attribution}
      src={custom.src}
      height={custom.height}
      width={custom.width}
      {...additionalProps}
    />,
  );
  shouldMatchSnapshot(
    'should render image with srcset correctly',
    <Component
      alt={landscape.alt}
      attribution={landscape.attribution}
      src={landscape.src}
      srcset={landscape.srcset}
      height={landscape.height}
      width={landscape.width}
    />,
  );
};

export default snapshotTests;
