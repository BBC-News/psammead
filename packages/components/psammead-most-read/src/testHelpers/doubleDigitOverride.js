import {
  GEL_SPACING_DBL,
  GEL_SPACING_TRPL,
  GEL_SPACING_QUAD,
  GEL_SPACING_QUIN,
  GEL_SPACING_SEXT,
  GEL_SPACING_SEPT,
} from '@bbc/gel-foundations/spacings';

// Services with fonts that have glyphs thinner than the majority of other fonts.
// This was mainly based on the old overrides (ie. Any group0 value < 2rem).
export const thinFontServices = [
  'pashto',
  'persian',
  'tamil',
  'telegu',
  'turkce',
  'urdu',
];

// If numberOfItems < 10, no extra spacing needs to be accounted for.
export const singleDigitDefault = {
  group0: GEL_SPACING_TRPL,
  group1: GEL_SPACING_TRPL,
  group2: GEL_SPACING_QUAD,
  group3: GEL_SPACING_QUIN,
  group5: GEL_SPACING_QUAD,
};

export const doubleDigitDefault = {
  group0: GEL_SPACING_QUIN,
  group1: GEL_SPACING_QUIN,
  group2: GEL_SPACING_SEXT,
  group3: GEL_SPACING_QUIN,
  group5: '3.9rem',
  group0_column: GEL_SPACING_QUAD,
  group1_column: GEL_SPACING_QUIN,
  group2_column: GEL_SPACING_SEXT,
  group3_5_column: '4rem',
};

export const singleDigitThin = {
  group0: GEL_SPACING_DBL,
  group1: GEL_SPACING_TRPL,
  group2: GEL_SPACING_TRPL,
  group3: GEL_SPACING_QUAD,
  group5: GEL_SPACING_QUAD,
};

export const doubleDigitThin = {
  group0: GEL_SPACING_DBL,
  group1: GEL_SPACING_TRPL,
  group2: GEL_SPACING_TRPL,
  group3: GEL_SPACING_TRPL,
  group5: GEL_SPACING_QUIN,
  group0_column: GEL_SPACING_QUAD,
  group1_column: GEL_SPACING_QUIN,
  group2_column: GEL_SPACING_SEXT,
  group3_5_column: GEL_SPACING_SEPT,
};
