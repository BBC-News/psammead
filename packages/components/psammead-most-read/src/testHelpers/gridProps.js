const mostReadListGridProps = {
  enableGelGutters: true,
  enableGelMargins: true,
  columns: {
    group0: 6,
    group1: 6,
    group2: 6,
    group3: 6,
    group4: 8,
    group5: 20,
  },
};

const mostReadItemGridProps = {
  item: true,
  columns: {
    group0: 6,
    group1: 6,
    group2: 3,
    group3: 3,
    group4: 4,
    group5: 4,
  },
};

export { mostReadListGridProps, mostReadItemGridProps };
