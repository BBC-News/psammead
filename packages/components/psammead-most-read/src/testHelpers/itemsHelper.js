import { TEXT_VARIANTS } from '@bbc/psammead-storybook-helpers';

export const getItem = service => {
  const baseUrl = 'https://www.bbc.com';
  const { text, articlePath } = TEXT_VARIANTS[service];
  return {
    title: text,
    href: `${baseUrl}${articlePath}`,
  };
};

export const getItems = (service = 'news') => Array(10).fill(getItem(service));
