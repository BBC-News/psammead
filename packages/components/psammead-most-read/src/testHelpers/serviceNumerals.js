import {
  Burmese,
  Bengali,
  EasternArabic,
  Nepali,
  WesternArabic,
} from '@bbc/psammead-locales/numerals';

const serviceNumerals = service => {
  const servicesNonWesternNumerals = {
    arabic: EasternArabic,
    bengali: Bengali,
    burmese: Burmese,
    nepali: Nepali,
    pashto: EasternArabic,
    persian: EasternArabic,
    urdu: EasternArabic,
  };
  return servicesNonWesternNumerals[service]
    ? servicesNonWesternNumerals[service]
    : WesternArabic;
};

export default serviceNumerals;
