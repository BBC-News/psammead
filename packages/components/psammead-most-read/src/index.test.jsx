import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { arabic, latin } from '@bbc/gel-foundations/scripts';
import { getItems } from './testHelpers';
import { MostRead } from '.';

describe('MostRead', () => {
  shouldMatchSnapshot(
    'should render with ltr most read with correct dir',
    <MostRead
      items={getItems({ service: 'news', arraySize: 10 })}
      script={latin}
      service="news"
      header="Most Read"
    />,
  );
  shouldMatchSnapshot(
    'should render with rtl most read with correct dir',
    <MostRead
      items={getItems({ service: 'arabic', arraySize: 10 })}
      script={arabic}
      service="arabic"
      header="الأكثر قراءة"
      dir="rtl"
    />,
  );
});
