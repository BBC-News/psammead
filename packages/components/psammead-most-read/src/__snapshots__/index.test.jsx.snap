// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MostRead should render with ltr most read with correct dir 1`] = `
.c13 {
  font-size: 1.125rem;
  line-height: 1.375rem;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-bottom: 0.5rem;
}

.c13:hover,
.c13:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c13:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c12 {
  padding-top: 0.375rem;
  padding-bottom: 1.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin: 0;
  padding: 0;
}

.c8 {
  position: relative;
}

.c11 {
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 300;
  font-style: normal;
  font-size: 2rem;
  line-height: 2.25rem;
  position: relative;
  color: #B80000;
  margin: 0;
  padding: 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  min-height: 2.75rem;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c4 {
  font-size: 1.125rem;
  line-height: 1.375rem;
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #222222;
  background-color: #FFFFFF;
  margin: 1rem 0;
  padding-right: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  position: relative;
  margin-top: 2rem;
}

.c1 {
  margin: 0;
  padding: 0;
}

.c6 {
  list-style-type: none;
  margin: 0;
  padding: 0;
  grid-auto-flow: column;
  grid-template-rows: repeat( 5,[col-start] auto [col-end] );
}

@supports (display:grid) {
  .c5 {
    display: grid;
    position: initial;
  }
}

@media (max-width:14.9375rem) {
  .c7 {
    width: calc(100%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c7 {
    width: calc(100%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c7 {
    width: calc(100%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c7 {
    width: calc(50%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c7 {
    width: calc(50%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:80rem) {
  .c7 {
    width: calc(20%);
    display: inline-block;
    vertical-align: top;
  }
}

@supports (display:grid) {
  .c7 {
    display: block;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c13 {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

@media (min-width:37.5rem) {
  .c13 {
    font-size: 1.5rem;
    line-height: 1.75rem;
  }
}

@supports (grid-template-columns:fit-content(200px)) {
  .c12 {
    padding-right: 0;
  }
}

@media (max-width:14.9375rem) {
  .c10 {
    min-width: 2rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c10 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c10 {
    min-width: 3rem;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c10 {
    min-width: auto;
  }
}

@media (min-width:80rem) {
  .c10 {
    min-width: auto;
  }
}

@media (max-width:14.9375rem) {
  .c14 {
    min-width: 2rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c14 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c14 {
    min-width: 3rem;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c14 {
    min-width: 4rem;
  }
}

@media (min-width:80rem) {
  .c14 {
    min-width: auto;
  }
}

@media (max-width:14.9375rem) {
  .c15 {
    min-width: 2rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c15 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c15 {
    min-width: 3rem;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c15 {
    min-width: auto;
  }
}

@media (min-width:80rem) {
  .c15 {
    min-width: 4.25rem;
  }
}

@media (max-width:14.9375rem) {
  .c16 {
    min-width: 2rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c16 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c16 {
    min-width: 3rem;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c16 {
    min-width: 4rem;
  }
}

@media (min-width:80rem) {
  .c16 {
    min-width: auto;
  }
}

@media (max-width:14.9375rem) {
  .c17 {
    min-width: 2rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c17 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c17 {
    min-width: 3rem;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c17 {
    min-width: auto;
  }
}

@media (min-width:80rem) {
  .c17 {
    min-width: auto;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c11 {
    font-size: 2.5rem;
    line-height: 2.75rem;
  }
}

@media (min-width:37.5rem) {
  .c11 {
    font-size: 3.5rem;
    line-height: 3.75rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1.5rem;
    line-height: 1.75rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    margin: 0;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    padding-right: 1rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    margin-top: 1.5rem;
  }
}

@media (max-width:37.4375rem) {
  .c6 {
    grid-template-rows: repeat( 10,[col-start] auto [col-end] );
  }
}

@media (min-width:80rem) {
  .c6 {
    grid-auto-flow: row;
    grid-template-rows: repeat( 5,[col-start] auto [col-end] );
  }
}

<section
  aria-labelledby="Most-Read"
  class=""
  role="region"
>
  <div
    class="c0"
  >
    <h2
      class="c1"
    >
      <div
        class="c2"
      >
        <span
          class="c3"
        >
          <span
            class="c4"
            dir="ltr"
            id="Most-Read"
          >
            Most Read
          </span>
        </span>
      </div>
    </h2>
  </div>
  <ol
    class="c5 c6"
    dir="ltr"
    role="list"
  >
    <li
      class="c7 c8"
      dir="ltr"
      role="listitem"
    >
      <div
        class="c9"
      >
        <div
          class="c10"
          dir="ltr"
        >
          <span
            class="c11"
          >
            1
          </span>
        </div>
        <div
          class="c12"
          dir="ltr"
        >
          <a
            class="c13"
            href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
          >
            Could a computer ever create better art than a human?
          </a>
        </div>
      </div>
    </li>
    <li
      class="c7 c8"
      dir="ltr"
      role="listitem"
    >
      <div
        class="c9"
      >
        <div
          class="c14"
          dir="ltr"
        >
          <span
            class="c11"
          >
            2
          </span>
        </div>
        <div
          class="c12"
          dir="ltr"
        >
          <a
            class="c13"
            href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
          >
            Could a computer ever create better art than a human?
          </a>
        </div>
      </div>
    </li>
    <li
      class="c7 c8"
      dir="ltr"
      role="listitem"
    >
      <div
        class="c9"
      >
        <div
          class="c10"
          dir="ltr"
        >
          <span
            class="c11"
          >
            3
          </span>
        </div>
        <div
          class="c12"
          dir="ltr"
        >
          <a
            class="c13"
            href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
          >
            Could a computer ever create better art than a human?
          </a>
        </div>
      </div>
    </li>
    <li
      class="c7 c8"
      dir="ltr"
      role="listitem"
    >
      <div
        class="c9"
      >
        <div
          class="c14"
          dir="ltr"
        >
          <span
            class="c11"
          >
            4
          </span>
        </div>
        <div
          class="c12"
          dir="ltr"
        >
          <a
            class="c13"
            href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
          >
            Could a computer ever create better art than a human?
          </a>
        </div>
      </div>
    </li>
    <li
      class="c7 c8"
      dir="ltr"
      role="listitem"
    >
      <div
        class="c9"
      >
        <div
          class="c15"
          dir="ltr"
        >
          <span
            class="c11"
          >
            5
          </span>
        </div>
        <div
          class="c12"
          dir="ltr"
        >
          <a
            class="c13"
            href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
          >
            Could a computer ever create better art than a human?
          </a>
        </div>
      </div>
    </li>
    <li
      class="c7 c8"
      dir="ltr"
      role="listitem"
    >
      <div
        class="c9"
      >
        <div
          class="c16"
          dir="ltr"
        >
          <span
            class="c11"
          >
            6
          </span>
        </div>
        <div
          class="c12"
          dir="ltr"
        >
          <a
            class="c13"
            href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
          >
            Could a computer ever create better art than a human?
          </a>
        </div>
      </div>
    </li>
    <li
      class="c7 c8"
      dir="ltr"
      role="listitem"
    >
      <div
        class="c9"
      >
        <div
          class="c17"
          dir="ltr"
        >
          <span
            class="c11"
          >
            7
          </span>
        </div>
        <div
          class="c12"
          dir="ltr"
        >
          <a
            class="c13"
            href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
          >
            Could a computer ever create better art than a human?
          </a>
        </div>
      </div>
    </li>
    <li
      class="c7 c8"
      dir="ltr"
      role="listitem"
    >
      <div
        class="c9"
      >
        <div
          class="c16"
          dir="ltr"
        >
          <span
            class="c11"
          >
            8
          </span>
        </div>
        <div
          class="c12"
          dir="ltr"
        >
          <a
            class="c13"
            href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
          >
            Could a computer ever create better art than a human?
          </a>
        </div>
      </div>
    </li>
    <li
      class="c7 c8"
      dir="ltr"
      role="listitem"
    >
      <div
        class="c9"
      >
        <div
          class="c17"
          dir="ltr"
        >
          <span
            class="c11"
          >
            9
          </span>
        </div>
        <div
          class="c12"
          dir="ltr"
        >
          <a
            class="c13"
            href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
          >
            Could a computer ever create better art than a human?
          </a>
        </div>
      </div>
    </li>
    <li
      class="c7 c8"
      dir="ltr"
      role="listitem"
    >
      <div
        class="c9"
      >
        <div
          class="c16"
          dir="ltr"
        >
          <span
            class="c11"
          >
            10
          </span>
        </div>
        <div
          class="c12"
          dir="ltr"
        >
          <a
            class="c13"
            href="https://www.bbc.com/news/articles/cn7k01xp8kxo"
          >
            Could a computer ever create better art than a human?
          </a>
        </div>
      </div>
    </li>
  </ol>
</section>
`;

exports[`MostRead should render with rtl most read with correct dir 1`] = `
.c13 {
  font-size: 1.5rem;
  line-height: 2rem;
  font-family: "BBC Nassim Arabic",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 700;
  font-style: normal;
  position: static;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-bottom: 0.5rem;
}

.c13:hover,
.c13:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c13:before {
  bottom: 0;
  content: '';
  left: 0;
  overflow: hidden;
  position: absolute;
  right: 0;
  top: 0;
  white-space: nowrap;
  z-index: 1;
}

.c12 {
  padding-top: 0.375rem;
  padding-bottom: 1.5rem;
  padding-right: 1rem;
  padding-left: 1rem;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin: 0;
  padding: 0;
}

.c8 {
  position: relative;
}

.c11 {
  font-family: "BBC Nassim Arabic",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 2rem;
  line-height: 2.25rem;
  position: relative;
  color: #B80000;
  margin: 0;
  padding: 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  min-height: 2.75rem;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c4 {
  font-size: 1.5rem;
  line-height: 2rem;
  font-family: "BBC Nassim Arabic",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #222222;
  background-color: #FFFFFF;
  margin: 1rem 0;
  padding-left: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  position: relative;
  margin-top: 2rem;
}

.c1 {
  margin: 0;
  padding: 0;
}

.c6 {
  list-style-type: none;
  margin: 0;
  padding: 0;
  grid-auto-flow: column;
  grid-template-rows: repeat( 5,[col-start] auto [col-end] );
}

@supports (display:grid) {
  .c5 {
    display: grid;
    position: initial;
  }
}

@media (max-width:14.9375rem) {
  .c7 {
    width: calc(100%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c7 {
    width: calc(100%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c7 {
    width: calc(100%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:37.5rem) and (max-width:62.9375rem) {
  .c7 {
    width: calc(50%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c7 {
    width: calc(50%);
    display: inline-block;
    vertical-align: top;
  }
}

@media (min-width:80rem) {
  .c7 {
    width: calc(20%);
    display: inline-block;
    vertical-align: top;
  }
}

@supports (display:grid) {
  .c7 {
    display: block;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c13 {
    font-size: 1.625rem;
    line-height: 2.25rem;
  }
}

@media (min-width:37.5rem) {
  .c13 {
    font-size: 1.75rem;
    line-height: 2.25rem;
  }
}

@supports (grid-template-columns:fit-content(200px)) {
  .c12 {
    padding-left: 0;
  }
}

@media (max-width:14.9375rem) {
  .c10 {
    min-width: 2.5rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c10 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c10 {
    min-width: 2.5rem;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c10 {
    min-width: auto;
  }
}

@media (min-width:80rem) {
  .c10 {
    min-width: auto;
  }
}

@media (max-width:14.9375rem) {
  .c14 {
    min-width: 2.5rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c14 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c14 {
    min-width: 2.5rem;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c14 {
    min-width: 4rem;
  }
}

@media (min-width:80rem) {
  .c14 {
    min-width: auto;
  }
}

@media (max-width:14.9375rem) {
  .c15 {
    min-width: 2.5rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c15 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c15 {
    min-width: 2.5rem;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c15 {
    min-width: auto;
  }
}

@media (min-width:80rem) {
  .c15 {
    min-width: 3.75rem;
  }
}

@media (max-width:14.9375rem) {
  .c16 {
    min-width: 2.5rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c16 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c16 {
    min-width: 2.5rem;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c16 {
    min-width: 4rem;
  }
}

@media (min-width:80rem) {
  .c16 {
    min-width: auto;
  }
}

@media (max-width:14.9375rem) {
  .c17 {
    min-width: 2.5rem;
  }
}

@media (min-width:15rem) and (max-width:24.9375rem) {
  .c17 {
    min-width: 2.5rem;
  }
}

@media (min-width:25rem) and (max-width:37.4375rem) {
  .c17 {
    min-width: 2.5rem;
  }
}

@media (min-width:37.5rem) and (max-width:80rem) {
  .c17 {
    min-width: auto;
  }
}

@media (min-width:80rem) {
  .c17 {
    min-width: auto;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c11 {
    font-size: 2.5rem;
    line-height: 2.75rem;
  }
}

@media (min-width:37.5rem) {
  .c11 {
    font-size: 3.5rem;
    line-height: 3.75rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 1.625rem;
    line-height: 2.25rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 1.75rem;
    line-height: 2.25rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    margin: 0;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    padding-left: 1rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    margin-top: 1.5rem;
  }
}

@media (max-width:37.4375rem) {
  .c6 {
    grid-template-rows: repeat( 10,[col-start] auto [col-end] );
  }
}

@media (min-width:80rem) {
  .c6 {
    grid-auto-flow: row;
    grid-template-rows: repeat( 5,[col-start] auto [col-end] );
  }
}

<section
  aria-labelledby="Most-Read"
  class=""
  role="region"
>
  <div
    class="c0"
  >
    <h2
      class="c1"
    >
      <div
        class="c2"
      >
        <span
          class="c3"
        >
          <span
            class="c4"
            dir="rtl"
            id="Most-Read"
          >
            الأكثر قراءة
          </span>
        </span>
      </div>
    </h2>
  </div>
  <ol
    class="c5 c6"
    dir="rtl"
    role="list"
  >
    <li
      class="c7 c8"
      dir="rtl"
      role="listitem"
    >
      <div
        class="c9"
      >
        <div
          class="c10"
          dir="rtl"
        >
          <span
            class="c11"
          >
            ۱
          </span>
        </div>
        <div
          class="c12"
          dir="rtl"
        >
          <a
            class="c13"
            href="https://www.bbc.com/arabic/articles/c1er5mjnznzo"
          >
            لماذا يخجل البعض من اسم قريته في مصر؟
          </a>
        </div>
      </div>
    </li>
    <li
      class="c7 c8"
      dir="rtl"
      role="listitem"
    >
      <div
        class="c9"
      >
        <div
          class="c14"
          dir="rtl"
        >
          <span
            class="c11"
          >
            ۲
          </span>
        </div>
        <div
          class="c12"
          dir="rtl"
        >
          <a
            class="c13"
            href="https://www.bbc.com/arabic/articles/c1er5mjnznzo"
          >
            لماذا يخجل البعض من اسم قريته في مصر؟
          </a>
        </div>
      </div>
    </li>
    <li
      class="c7 c8"
      dir="rtl"
      role="listitem"
    >
      <div
        class="c9"
      >
        <div
          class="c10"
          dir="rtl"
        >
          <span
            class="c11"
          >
            ۳
          </span>
        </div>
        <div
          class="c12"
          dir="rtl"
        >
          <a
            class="c13"
            href="https://www.bbc.com/arabic/articles/c1er5mjnznzo"
          >
            لماذا يخجل البعض من اسم قريته في مصر؟
          </a>
        </div>
      </div>
    </li>
    <li
      class="c7 c8"
      dir="rtl"
      role="listitem"
    >
      <div
        class="c9"
      >
        <div
          class="c14"
          dir="rtl"
        >
          <span
            class="c11"
          >
            ۴
          </span>
        </div>
        <div
          class="c12"
          dir="rtl"
        >
          <a
            class="c13"
            href="https://www.bbc.com/arabic/articles/c1er5mjnznzo"
          >
            لماذا يخجل البعض من اسم قريته في مصر؟
          </a>
        </div>
      </div>
    </li>
    <li
      class="c7 c8"
      dir="rtl"
      role="listitem"
    >
      <div
        class="c9"
      >
        <div
          class="c15"
          dir="rtl"
        >
          <span
            class="c11"
          >
            ۵
          </span>
        </div>
        <div
          class="c12"
          dir="rtl"
        >
          <a
            class="c13"
            href="https://www.bbc.com/arabic/articles/c1er5mjnznzo"
          >
            لماذا يخجل البعض من اسم قريته في مصر؟
          </a>
        </div>
      </div>
    </li>
    <li
      class="c7 c8"
      dir="rtl"
      role="listitem"
    >
      <div
        class="c9"
      >
        <div
          class="c16"
          dir="rtl"
        >
          <span
            class="c11"
          >
            ۶
          </span>
        </div>
        <div
          class="c12"
          dir="rtl"
        >
          <a
            class="c13"
            href="https://www.bbc.com/arabic/articles/c1er5mjnznzo"
          >
            لماذا يخجل البعض من اسم قريته في مصر؟
          </a>
        </div>
      </div>
    </li>
    <li
      class="c7 c8"
      dir="rtl"
      role="listitem"
    >
      <div
        class="c9"
      >
        <div
          class="c17"
          dir="rtl"
        >
          <span
            class="c11"
          >
            ۷
          </span>
        </div>
        <div
          class="c12"
          dir="rtl"
        >
          <a
            class="c13"
            href="https://www.bbc.com/arabic/articles/c1er5mjnznzo"
          >
            لماذا يخجل البعض من اسم قريته في مصر؟
          </a>
        </div>
      </div>
    </li>
    <li
      class="c7 c8"
      dir="rtl"
      role="listitem"
    >
      <div
        class="c9"
      >
        <div
          class="c16"
          dir="rtl"
        >
          <span
            class="c11"
          >
            ۸
          </span>
        </div>
        <div
          class="c12"
          dir="rtl"
        >
          <a
            class="c13"
            href="https://www.bbc.com/arabic/articles/c1er5mjnznzo"
          >
            لماذا يخجل البعض من اسم قريته في مصر؟
          </a>
        </div>
      </div>
    </li>
    <li
      class="c7 c8"
      dir="rtl"
      role="listitem"
    >
      <div
        class="c9"
      >
        <div
          class="c17"
          dir="rtl"
        >
          <span
            class="c11"
          >
            ۹
          </span>
        </div>
        <div
          class="c12"
          dir="rtl"
        >
          <a
            class="c13"
            href="https://www.bbc.com/arabic/articles/c1er5mjnznzo"
          >
            لماذا يخجل البعض من اسم قريته في مصر؟
          </a>
        </div>
      </div>
    </li>
    <li
      class="c7 c8"
      dir="rtl"
      role="listitem"
    >
      <div
        class="c9"
      >
        <div
          class="c16"
          dir="rtl"
        >
          <span
            class="c11"
          >
            ۱۰
          </span>
        </div>
        <div
          class="c12"
          dir="rtl"
        >
          <a
            class="c13"
            href="https://www.bbc.com/arabic/articles/c1er5mjnznzo"
          >
            لماذا يخجل البعض من اسم قريته في مصر؟
          </a>
        </div>
      </div>
    </li>
  </ol>
</section>
`;
